### ISSUE TITLE
[Bug]: 2.2.9:  "enableButton" Property Is not being honored 

### ISSUE THREAD

**Issue by gmogoody** [homebridge-eufy-security/plugin#401]
### What happened?

After updating and adding all my devices as unbridged. I noticed that the "motionButton" property is working as expected but devices like the indoor Pan & Tilt that also have an "enableButton" property no longer works. If property is "true" no accessory is added in HomeBridge according to the GUI accessory tab. 

### Device Type

Indoor Camera (Supported)

### Plugin Version

v2.0.x (Supported)

### HomeBridge Version

v1.6.x (Supported)

### NodeJS Version

v18 (Supported)

### Operating System

Raspbian 10+ (Supported)

### Relevant log output

```shell
None
```

### config

```shell
{
    "enableDetailedLogging": false,
    "enableCamera": false,
    "CameraMaxLivestreamDuration": 30,
    "pollingIntervalMinutes": 10,
    "hkHome": 1,
    "hkAway": 0,
    "hkNight": 3,
    "hkOff": 63,
    "ignoreStations": [
        "T8021P1020430F46",
        "T8410P4122295507",
        "T8410P412229582D",
        "T8410P4122294D21",
        "T8410P4223020079"
    ],
    "ignoreDevices": [
        "T8510Q11213401BA",
        "T8140P61210224BF",
        "T8210P0020210DFC"
    ],
    "country": "US",
    "ffmpegdebug": false,
    "cameras": [
        {
            "serialNumber": "T8510Q11213401BA",
            "rtsp": false,
            "talkback": false
        },
        {
            "serialNumber": "T8140P61210224BF",
            "rtsp": false
        },
        {
            "serialNumber": "T8140P012052024D",
            "rtsp": false,
            "hsv": false,
            "hsvRecordingDuration": 90,
            "hsvConfig": {},
            "snapshotHandlingMethod": 3,
            "enableButton": false,
            "motionButton": true,
            "unbridge": true
        },
        {
            "serialNumber": "T8210P0020210DFC",
            "rtsp": false,
            "enableCamera": true,
            "hsv": false,
            "hsvRecordingDuration": 90,
            "hsvConfig": {},
            "snapshotHandlingMethod": 3
        },
        {
            "serialNumber": "T8410P4122295507",
            "enableButton": true,
            "motionButton": false,
            "unbridge": true
        },
        {
            "serialNumber": "T8410P412229582D",
            "enableButton": true,
            "motionButton": false,
            "unbridge": true
        },
        {
            "serialNumber": "T8410P4122294D21",
            "enableButton": true,
            "motionButton": false,
            "unbridge": true
        },
        {
            "serialNumber": "T8410P4223020079",
            "enableButton": true,
            "motionButton": false,
            "unbridge": true
        },
        {
            "serialNumber": "T8160P112314317E",
            "enableButton": false,
            "motionButton": true,
            "unbridge": true
        }
    ],
    "cleanCache": false,
    "unbridge": true,
    "username": "",
    "password": "",
    "deviceName": "",
    "platform": "EufySecurity",
    "_bridge": {
        "username": "0E:B2:E1:79:A0:28",
        "port": 45498
    },
    "stations": [
        {
            "serialNumber": "T8030P13223307C6",
            "manualTriggerModes": [
                0,
                3
            ]
        },
        {
            "serialNumber": "T8030P1322330CF1",
            "manualTriggerModes": [
                0,
                3
            ]
        }
    ]
}
```

**Comment by gmogoody** [comment]
![Settings](https://github.com/homebridge-eufy-security/plugin/assets/35224562/951bcfd2-1252-41d5-b54b-e4301ab0fe7d)
![HB_GUI](https://github.com/homebridge-eufy-security/plugin/assets/35224562/517cbfec-175f-494a-9f1f-f978785ecdb3)

**Comment by lenoxys** [comment]
can you provide me logs? You should look for REGISTER debug line. Here an example of mine : 

Enable button disabled as per config :
```
[09/03/2023 16:33:49]   [EufySecurity-2.2.10]   DEBUG   [CameraAccessory.ts:86] Doorbell Enabled config: true
[09/03/2023 16:33:49]   [EufySecurity-2.2.10]   DEBUG   [CameraAccessory.ts:89] Doorbell has a enabled, so append switchEnabled characteristic to it.
[09/03/2023 16:33:49]   [EufySecurity-2.2.10]   DEBUG   [BaseAccessory.ts:124]  Doorbell REGISTER SERVICE 'Switch / On': 00000025-0000-1000-8000-0026BB765291
```

Motion button disabled as per config :
```
[09/03/2023 16:33:49]   [EufySecurity-2.2.10]   DEBUG   [CameraAccessory.ts:86] Doorbell Motion config: false
[09/03/2023 16:33:49]   [EufySecurity-2.2.10]   DEBUG   [CameraAccessory.ts:93] Doorbell Looks like not compatible with motionDetection or this has been disabled within configuration
```

Light enabled but doesn't exist on this doorbell : 
```
[09/03/2023 16:33:49]   [EufySecurity-2.2.10]   DEBUG   [CameraAccessory.ts:86] Doorbell Light config: true
[09/03/2023 16:33:49]   [EufySecurity-2.2.10]   DEBUG   [CameraAccessory.ts:93] Doorbell Looks like not compatible with light or this has been disabled within configuration
```

**Comment by gmogoody** [comment]
After Upgrade to 2.2.10. Still see issue

I see that for My HomeBases
{code}
[09/04/2023 14:02:47]   [EufySecurity-2.2.10]   DEBUG   [BaseAccessory.ts:124]  Office HomeBase REGISTER SERVICE 'Switch / On': 00000025-0000-1000-8000-0026BB765291
{code}

The Backyard S330 Battery Camera. This exposes both a Motion On/Off and a Spotlight Switch
{code}
[09/04/2023 14:02:48]   [EufySecurity-2.2.10]   DEBUG   [CameraAccessory.ts:50] Backyard Constructed Camera
[09/04/2023 14:02:48]   [EufySecurity-2.2.10]   DEBUG   [CameraAccessory.ts:62] Backyard has a motion sensor.
[09/04/2023 14:02:48]   [EufySecurity-2.2.10]   DEBUG   [BaseAccessory.ts:124]  Backyard REGISTER SERVICE 'MotionSensor / Motion Detected': 00000022-0000-1000-8000-0026BB765291
[Backyard@Backyard@Battery Level] Characteristic not in required or optional characteristic section for service MotionSensor. Adding anyway.
[09/04/2023 14:02:48]   [EufySecurity-2.2.10]   DEBUG   [BaseAccessory.ts:124]  Backyard REGISTER SERVICE 'MotionSensor / Battery Level': 00000068-0000-1000-8000-0026BB765291
[09/04/2023 14:02:48]   [EufySecurity-2.2.10]   INFO    [platform.ts:603]       Publishing unbridged accessory externally: Backyard
{code}

The Garage Battery Camera. This exposes a Motion On/Off
{code}
[09/04/2023 14:02:48]   [EufySecurity-2.2.10]   DEBUG   [CameraAccessory.ts:62] Garage has a motion sensor.
[09/04/2023 14:02:48]   [EufySecurity-2.2.10]   DEBUG   [BaseAccessory.ts:124]  Garage REGISTER SERVICE 'MotionSensor / Motion Detected': 00000022-0000-1000-8000-0026BB765291
[Garage@Garage@Battery Level] Characteristic not in required or optional characteristic section for service MotionSensor. Adding anyway.
[09/04/2023 14:02:48]   [EufySecurity-2.2.10]   DEBUG   [BaseAccessory.ts:124]  Garage REGISTER SERVICE 'MotionSensor / Battery Level': 00000068-0000-1000-8000-0026BB765291
[09/04/2023 14:02:48]   [EufySecurity-2.2.10]   INFO    [platform.ts:603]       Publishing unbridged accessory externally: Garage
{code}

All My Indoor Pan & Tilt Cameras Logs are similar to this.
{code}
[09/04/2023 14:02:48]   [EufySecurity-2.2.10]   DEBUG   [CameraAccessory.ts:62] Dining Room has a motion sensor.
[09/04/2023 14:02:48]   [EufySecurity-2.2.10]   DEBUG   [BaseAccessory.ts:124]  Dining Room REGISTER SERVICE 'MotionSensor / Motion Detected': 00000022-0000-1000-8000-0026BB765291
[09/04/2023 14:02:48]   [EufySecurity-2.2.10]   DEBUG   [Device.ts:80]  Dining Room has no battery
[09/04/2023 14:02:48]   [EufySecurity-2.2.10]   INFO    [platform.ts:603]       Publishing unbridged accessory externally: Dining Room
{code}

**Comment by lenoxys** [comment]
Is it on purpose that you set 

```
        {
            "serialNumber": "T8410P4122295507",
            "enableButton": true,
            "motionButton": false,
            "unbridge": true
        },
        {
            "serialNumber": "T8410P412229582D",
            "enableButton": true,
            "motionButton": false,
            "unbridge": true
        },
        {
            "serialNumber": "T8410P4122294D21",
            "enableButton": true,
            "motionButton": false,
            "unbridge": true
        },
        {
            "serialNumber": "T8410P4223020079",
            "enableButton": true,
            "motionButton": false,
            "unbridge": true
        },
        {
            "serialNumber": "T8160P112314317E",
            "enableButton": false,
            "motionButton": true,
            "unbridge": true
        }
```
As motion detector instead of camera ? 

This would explain why. Button are exposed when camera selected not motion detected. What's your use case to have button when it's only motion detector ?

**Comment by gmogoody** [comment]
Using these setting were not limited to setting the device as "Camera" before 2.2.6

I have all my devices setup as Motions, but still exposed either the "enableButton" or "motionButton". 

I use these switches in automations. For example I have HK Turn Off the Indoor Wired Cameras with "enableButton" when someone is home. 

I never used the Plugin as "Cameras" due to the limitations and having to use the Cloud and thus one of the two streams. If Eufy allowed more streams I would have used the plugin

**Comment by lenoxys** [comment]
Will expose it in the next patch

**Comment by gmogoody** [comment]
Thanks so much for the support. Issue is fixed in 2.2.12
